### ISSUE TITLE
Nonlinear Boundary Value problems not accepting real number as index

### ISSUE THREAD

**Issue by olivariusFR** [JuliaApproximation/ApproxFun.jl#704]
I am using JuliaApproximation to solve the Lane-Emden equation.
Following the example of Nonlinear Boundary Value problems for Solving  a nonlinear boundary value problem satisfying the ODE with boundary conditions 

It’s working great for integer index n. 
```
x  = Fun(identity, 0..10)
n = 2
u₀ = 0.0x # initial guess
N = u -> [u'(0), u(0)-1, x*u'' + 2*u' + x*u^n]
u = newton(N, u₀) # perform Newton iteration in function space
plot(u)
```

When I try for n as a real number, n = 4.5 for example, I get the following error:
```
MethodError: no method matching ^(::ApproxFun.DualFun{Fun{Chebyshev{Interval{:closed,:closed,Int64},Float64},Float64,Array{Float64,1}},ConstantOperator{Float64,Chebyshev{Interval{:closed,:closed,Int64},Float64}}}, ::Float64)
Closest candidates are:
  ^(!Matched::Missing, ::Number) at missing.jl:115
  ^(!Matched::Float64, ::Float64) at math.jl:870
  ^(!Matched::Irrational{:ℯ}, ::Number) at mathconstants.jl:91
  ...

Stacktrace:
 [1] (::var"#65#66")(::ApproxFun.DualFun{Fun{Chebyshev{Interval{:closed,:closed,Int64},Float64},Float64,Array{Float64,1}},ConstantOperator{Float64,Chebyshev{Interval{:closed,:closed,Int64},Float64}}}) at ./In[49]:6
 [2] newton(::var"#65#66", ::Fun{Chebyshev{Interval{:closed,:closed,Int64},Float64},Float64,Array{Float64,1}}; maxiterations::Int64, tolerance::Float64) at /projects/d1d068b5-3a9d-4e16-9252-52734dea2c8e/.julia/packages/ApproxFun/IC3js/src/Extras/autodifferentiation.jl:98
 [3] newton(::Function, ::Fun{Chebyshev{Interval{:closed,:closed,Int64},Float64},Float64,Array{Float64,1}}) at /projects/d1d068b5-3a9d-4e16-9252-52734dea2c8e/.julia/packages/ApproxFun/IC3js/src/Extras/autodifferentiation.jl:95
 [4] top-level scope at In[49]:7
```

**Comment by dlfivefifty** [comment]
Fixed on master, will tag once tests pass

**Comment by dlfivefifty** [comment]
Please note that you can't (right now) use a 0 initial guess, so try 0.0x+1

**Comment by olivariusFR** [comment]
0.0x was working on the version I had.

Thank you for the fix
I will update as soon as it's available

**Comment by olivariusFR** [comment]
Not sure it's related but the solver has a problem at n=2 and n=0.
It seems to be stuck for n=2 (no convergence)
And there is an error at n=0:
`Implement Multiplication(::Fun{PiecewiseSpace{Tuple{JacobiWeight{Chebyshev{Interval{:closed,:closed,Float64},Float64},Interval{:closed,:closed,Float64},Float64,Int64},JacobiWeight{Chebyshev{Interval{:closed,:closed,Float64},Float64},Interval{:closed,:closed,Float64},Float64,Int64}},DomainSets.UnionDomain{Float64,Tuple{Interval{:closed,:closed,Float64},Interval{:closed,:closed,Float64}}},Float64}},::Chebyshev{Interval{:closed,:closed,Int64},Float64})

Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] defaultMultiplication(::Fun{PiecewiseSpace{Tuple{JacobiWeight{Chebyshev{Interval{:closed,:closed,Float64},Float64},Interval{:closed,:closed,Float64},Float64,Int64},JacobiWeight{Chebyshev{Interval{:closed,:closed,Float64},Float64},Interval{:closed,:closed,Float64},Float64,Int64}},DomainSets.UnionDomain{Float64,Tuple{Interval{:closed,:closed,Float64},Interval{:closed,:closed,Float64}}},Float64},Float64,Array{Float64,1}}, ::Chebyshev{Interval{:closed,:closed,Int64},Float64}) at /projects/d1d068b5-3a9d-4e16-9252-52734dea2c8e/.julia/packages/ApproxFunBase/Jv4NR/src/Operators/banded/Multiplication.jl:34
 [3] Multiplication(::Fun{PiecewiseSpace{Tuple{JacobiWeight{Chebyshev{Interval{:closed,:closed,Float64},Float64},Interval{:closed,:closed,Float64},Float64,Int64},JacobiWeight{Chebyshev{Interval{:closed,:closed,Float64},Float64},Interval{:closed,:closed,Float64},Float64,Int64}},DomainSets.UnionDomain{Float64,Tuple{Interval{:closed,:closed,Float64},Interval{:closed,:closed,Float64}}},Float64},Float64,Array{Float64,1}}, ::Chebyshev{Interval{:closed,:closed,Int64},Float64}) at /projects/d1d068b5-3a9d-4e16-9252-52734dea2c8e/.julia/packages/ApproxFunBase/Jv4NR/src/Operators/banded/Multiplication.jl:39
 [4] *(::Fun{PiecewiseSpace{Tuple{JacobiWeight{Chebyshev{Interval{:closed,:closed,Float64},Float64},Interval{:closed,:closed,Float64},Float64,Int64},JacobiWeight{Chebyshev{Interval{:closed,:closed,Float64},Float64},Interval{:closed,:closed,Float64},Float64,Int64}},DomainSets.UnionDomain{Float64,Tuple{Interval{:closed,:closed,Float64},Interval{:closed,:closed,Float64}}},Float64},Float64,Array{Float64,1}}, ::ConstantOperator{Float64,Chebyshev{Interval{:closed,:closed,Int64},Float64}}) at /projects/d1d068b5-3a9d-4e16-9252-52734dea2c8e/.julia/packages/ApproxFunBase/Jv4NR/src/Operators/general/algebra.jl:570
 [5] *(::Int64, ::Function, ::ConstantOperator{Float64,Chebyshev{Interval{:closed,:closed,Int64},Float64}}) at ./operators.jl:529
 [6] ^(::ApproxFun.DualFun{Fun{Chebyshev{Interval{:closed,:closed,Int64},Float64},Float64,Array{Float64,1}},ConstantOperator{Float64,Chebyshev{Interval{:closed,:closed,Int64},Float64}}}, ::Int64) at /projects/d1d068b5-3a9d-4e16-9252-52734dea2c8e/.julia/packages/ApproxFun/IC3js/src/Extras/autodifferentiation.jl:29
 [7] (::var"#223#224")(::ApproxFun.DualFun{Fun{Chebyshev{Interval{:closed,:closed,Int64},Float64},Float64,Array{Float64,1}},ConstantOperator{Float64,Chebyshev{Interval{:closed,:closed,Int64},Float64}}}) at ./In[262]:6
 [8] newton(::var"#223#224", ::Fun{Chebyshev{Interval{:closed,:closed,Int64},Float64},Float64,Array{Float64,1}}; maxiterations::Int64, tolerance::Float64) at /projects/d1d068b5-3a9d-4e16-9252-52734dea2c8e/.julia/packages/ApproxFun/IC3js/src/Extras/autodifferentiation.jl:98
 [9] newton(::Function, ::Fun{Chebyshev{Interval{:closed,:closed,Int64},Float64},Float64,Array{Float64,1}}) at /projects/d1d068b5-3a9d-4e16-9252-52734dea2c8e/.julia/packages/ApproxFun/IC3js/src/Extras/autodifferentiation.jl:95
 [10] top-level scope at In[262]:7
`

**Comment by olivariusFR** [comment]
For info, here is the results with chebfun (Matlab)
![image](https://user-images.githubusercontent.com/58397351/82502493-15dbcc00-9aac-11ea-987b-0f42c850f22b.png)
https://www.chebfun.org/examples/ode-nonlin/LaneEmden.html
